homeassistant/components/shelly/__init__.py

```diff
@@ -144,16 +182,47 @@ def __init__(self, hass, entry, device: aioshelly.Device):
 
         self.device.subscribe_updates(self.async_set_updated_data)
 
-        self._async_remove_input_events_handler = self.async_add_listener(
-            self._async_input_events_handler
+        self._async_remove_device_updates_handler = self.async_add_listener(
+            self._async_device_updates_handler
         )
         self._last_input_events_count = dict()
 
         hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self._handle_ha_stop)
 
     @callback
-    def _async_input_events_handler(self):
-        """Handle device input events."""
+    def _async_device_updates_handler(self):
+        """Handle device updates."""
+        if not self.device.initialized:
+            return
+
+        # Restored device initialized, update entities
```

Comment:
If this is for initialization, do it in the constructor? Seems weird to do it inside the generic updatae handler.

Reply:
Removed from here will restore inside entity